### PR TITLE
gpt-4 adapter created

### PR DESCRIPTION
**Pull request by cxalem**
## DESCRIPTION OF THE PULL REQUEST

This is a pretty simple GPT-4 text completions adapter. The next step is to create an endpoint capable of returning the data stream using Elysia!

## CHANGES MADE 

- [x] adds a new adapter
- [ ] creates a new endpoint
- [x] adds a relevant test

## TESTINGS DONE

1. user make a request to the OpenAi API
2. user gets a text completion response

## CHECKLIST
- [x] I have read the [CONTRIBUTING.md]
- [x] I have followed the [CODE_OF_CONDUCT.md]
- [x] I have written tests

### CODE REVIEW

File: src/infrastructure/adaptaters/gpt4Adapter.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+const openai = new OpenAI({
+  apiKey: process.env.OPENAI_API_KEY,
+});
+
+type gpt4Adapter = (
+  messages: string,
+  SYSTEM_PROMPT: string
+) => Promise<Stream<ChatCompletionChunk> | Response>;
+
+export const gpt4Adapter: gpt4Adapter = async (messages, SYSTEM_PROMPT) => {
```
Comment (GuiBibeau) [inline src/infrastructure/adaptaters/gpt4Adapter.ts:15]: why use constant casing?

File: package.json [hunk 2]
```diff
@@ -10,11 +10,13 @@
   },
   "dependencies": {
     "@elysiajs/bearer": "^0.7.0",
+    "dotenv": "^16.4.0",
```
Comment (GuiBibeau) [inline package.json:13]: no need for dotenv with bun:

https://bun.sh/guides/runtime/read-env
https://bun.sh/guides/runtime/set-env

File: package.json [hunk 3]
```diff
@@ -10,11 +10,13 @@
...
     "elysia": "latest",
     "ioredis": "^5.3.2",
     "jsonwebtoken": "8.5.1",
     "langchain": "^0.0.197-rc.1",
     "neo4j-driver-lite": "^5.16.0",
+    "openai": "^4.25.0",
```
Comment (GuiBibeau) [inline package.json:19]: do you think we could use the HTTP API to avoid the dependencies that come with the package?

**Review by GuiBibeau — COMMENTED** [review]
Thank you for the contribution! I left a few comments

File: src/infrastructure/adaptaters/gpt4Adapter.ts (hunk 1 above)
Comment (cxalem) [reply src/infrastructure/adaptaters/gpt4Adapter.ts:15]: For no particular reason, mostly habit. Are there any standard we should follow here?

File: package.json (hunk 2 above)
Comment (cxalem) [reply package.json:13]: Awesome, will remove de dep!

File: package.json (hunk 3 above)
Comment (cxalem) [reply package.json:19]: Sure! I will update it to the HTTP API!

**Review by GuiBibeau — APPROVED** [review]
beautiful!

**Pull request merged**